testkit/testkit.go

```diff
@@ -254,6 +253,16 @@ func (tk *TestKit) HasPlan(sql string, plan string, args ...interface{}) bool {
 	return false
 }
 
+// MustHavePlan checks if the result execution plan contains specific plan.
+func (tk *TestKit) MustHavePlan(sql string, plan string, args ...interface{}) {
+	tk.require.True(tk.hasPlan(sql, plan, args...), fmt.Sprintf("%s doesn't have plan %s", sql, plan))
+}
+
+// MustNotHavePlan checks if the result execution plan contains specific plan.
+func (tk *TestKit) MustNotHavePlan(sql string, plan string, args ...interface{}) {
+	tk.require.False(tk.hasPlan(sql, plan, args...), fmt.Sprintf("%s shouldn't have plan %s", sql, plan))
+}
+
```

Comment:
If rs.rows were returned as well as true/false, then the full plan could be included in the test failure message.

Also the args part is only used in case the MustQuery fails, not if the true/false assertion fails, which I would expect.